src/crawlers/cheerio_crawler.js

```diff
@@ -465,6 +465,9 @@ class CheerioCrawler {
                 return JSON.parse(jsonString);
             },
             get body() {
+                // TODO (JC): What's the point of generating body dynamically here from DOM?
+                //  If somebody changes the DOM using Cheerio, the body will change too... this is quite weird and unexpected.
+                //  Why don't we just return the original string as we got it???
```

Comment:
Memory consumption. In high concurrency scenarios, having all the HTML strings in memory (as well as the object trees) for the full lifecycle of `handlePageFunction` was causing the crawler to scale down or even crash. It was pretty much unusable with less than 1GB of memory.

Reply:
Fair enough, this would make a very nice comment :)

Reply:
I had this problem when using `load` directly on the SERP HTML dump testing, had to rely on the parser as well